src/Compilers/CSharp/Portable/Symbols/AliasSymbol.cs

```diff
@@ -393,6 +393,14 @@ private NamespaceSymbol ResolveExternAliasTarget(BindingDiagnosticBag diagnostic
                 this.CheckUnsafeModifier(DeclarationModifiers.Unsafe, usingDirective.UnsafeKeyword.GetLocation(), diagnostics);
                 flags |= BinderFlags.UnsafeRegion;
             }
+            else
+            {
+                // Prior to C#12, allow the alias to be an unsafe region.  This allows us to maintain compat with prior
+                // versions of the compiler that allowed `using X = List<int*[]>` to be written.  In 12.0 and onwards
+                // though, we require the code to explicitly contain the `unsafe` keyword.
+                if (((CSharpParseOptions)syntax.SyntaxTree.Options).LanguageVersion < LanguageVersion.Preview)
+                    flags |= BinderFlags.UnsafeRegion;
```

Comment:
>flags |= BinderFlags.UnsafeRegion; [](http://example.com/codeflow?start=20&length=34)

Do we have tests for the "problematic" alias and its usage in a "safe" context?

Reply:
I'll add an explicit one to be sure.

Reply:
Added explicit tests to UsingAliasTests.